accession-commons-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/services/JpaBasicSpringDataRepositoryDatabaseServiceTest.java

```diff
@@ -292,16 +293,21 @@ public void testDeprecateOneVersion() throws AccessionDoesNotExistException, Acc
         final List<TestInactiveAccessionEntity> deprecated = inactiveRepository.findAllByHistoryId(entity.getId());
         assertEquals(1, deprecated.size());
         assertEquals("something2", deprecated.get(0).getValue());
+
+        try {
```

Comment:
You can use `@Test(expected = AccessionDeprecatedException.class)`

Reply:
yes but we will not be sure which statement thrown that error. so used try catch.

Reply:
Then you can use the ExpectedException rule as in [this tutorial](http://baddotrobot.com/blog/2012/03/27/expecting-exception-with-junit-rule/).